packages/promotion/src/utils/compute-actions/line-items.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  ApplicationMethodAllocationValues,
+  PromotionTypes,
+} from "@medusajs/types"
+import {
+  ApplicationMethodAllocation,
+  ApplicationMethodTargetType,
+  ComputedActions,
+  MedusaError,
+  calculateAdjustmentAmountFromPromotion,
+} from "@medusajs/utils"
+import { areRulesValidForContext } from "../validations"
+import { computeActionForBudgetExceeded } from "./usage"
+
+function validateContext(
+  contextKey: string,
+  context: PromotionTypes.ComputeActionContext[ApplicationMethodTargetType]
+) {
+  if (!context) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      `"${contextKey}" should be present as an array in the context for computeActions`
+    )
+  }
+}
+
+export function getComputedActionsForItems(
```

Comment:
It feels like the logic can just be here rather than creating a new function, since all we do here is just call validateContext before we call the actual function that holds the logic. I leave it up to you, it just feels like an unnecessary additional step